libraries/stdlib/osl/stdosl.h

```diff
@@ -573,6 +573,169 @@ closure color volume_matte() BUILTIN;
+// to 0.0 for disabling a lobe.
+// The closure may be vertically layered over a base BSDF for the surface beneath the
+// dielectric layer. This is done using the layer() closure. By chaining multiple 
+// dielectric_bsdf closures you can describe a surface with multiple specular lobes.
+// If transmission is enabled (transmission_tint > 0.0) the closure may be layered over
+// a VDF closure describing the surface interior to handle absorption and scattering
+// inside the medium.
+//
+closure color dielectric_bsdf(normal N, vector U, color reflection_tint, color transmission_tint, float roughness_x, float roughness_y, float ior, string distribution) BUILTIN;
+​
+// Constructs a reflection BSDF based on a microfacet reflectance model.
+// Uses a Fresnel curve with complex refraction index for conductors/metals.
+// If an artistic parametrization is preferred the artistic_ior() utility function
+// can be used to convert from artistic to physical parameters.
+//
+closure color conductor_bsdf(normal N, vector U, float roughness_x, float roughness_y, color ior, color extinction, string distribution) BUILTIN;
+​
+// Constructs a reflection and/or transmission BSDF based on a microfacet reflectance model
+// and a generalized Schlick Fresnel curve. The two tint parameters control the contribution
+// of each reflection/transmission lobe.
+// The closure may be vertically layered over a base BSDF for the surface beneath the
+// dielectric layer. This is done using the layer() closure. By chaining multiple 
+// dielectric_bsdf closures you can describe a surface with multiple specular lobes.
+// If transmission is enabled (transmission_tint > 0.0) the closure may be layered over
+// a VDF closure describing the surface interior to handle absorption and scattering
+// inside the medium.
+//
+// TODO:
+// - Transmission handling for this node has not been fully defined yet in MaterialX.
+//   In particular how is IOR for refractions derived from the f0, f90 parameterization?
+//   Do we just derive it from f0? For an artist it seems hard to control refractions that way.
+//
+closure color generalized_schlick_bsdf(normal N, vector U, color reflection_tint, color transmission_tint, float roughness_x, float roughness_y, color f0, color f90, float exponent, string distribution) BUILTIN;
+​
+// Constructs a translucent (diffuse transmission) BSDF based on the Lambert reflectance model.
+//
+closure color translucent_bsdf(normal N, color albedo) BUILTIN;
+
+// Constructs a closure that represents straight transmission through a surface.
+//
+// NOTE:
+//  - This is not a node in the MaterialX library, but the surface shader constructor
+//    node has an 'opacity' parameter to control textured cutout opacity.
+//
+closure color transparent_bsdf() BUILTIN;
+​
+// Constructs a BSSRDF for subsurface scattering within a homogeneous medium.
+//
+// TODO:
+// - Do we want/need this dedicated BSSRDF closure? We have had discussions before about
```

Comment:
One thing to consider is that depending on the random walk implementation (especially the boundary handling) - the albedo mapping to volume parameters can be very different.

To me that suggests that the albedo mapping should happen inside the renderer which in turn suggests there should be a dedicated closure for subsurface that can capture the intended "artist friendly" albedo/radius parameter and let the renderer infer the volume parameters that will produce that effect.

Reply:
Good points, we can keep this in if it makes sense. On MaterialX side we will keep this in as a node in the library anyway, there was a request from ILM to keep this in as a artist friendly construct to add SSS. So as long as this can be translated to OSL one way or another we are happy on our side.

Reply:
The current SSS radius parameter is essentially the desired mean-free-path (MFP) in each color channel. This isn’t really directly related to the rendered color I think, and generally it's quite unintuitive to have a color parameter whose channels contain lengths. (While the provided "albedo" is a vector of dimensionless numbers giving the per-channel scattering albedo, so more easily interpretable as a genuine color).

It could be better to specify a single "depth" which gives the MFP (explicitly in units of scene length), and then an absorption color whose meaning is “the transmittance of a slab of medium of that thickness”. This is the color you would see if white light was transmitted through a 1 MFP thick slab of medium. Then the specified extinction is taken to be -log(absorption color) / MFP.

That is the scheme we use in Arnold's standard surface and standard volume, i.e. there is a transmission_depth and a transmission_color, which are converted into the RGB extinction in the way described.

Reply:
@portsmouth is right, and this is probably my fault, since I defined it thusly in the spiritual predecessor to the Autodesk Standard Surface. In my poor old head, it kinda made sense to just have the mean free paths scaled by the color, but I think my "mental intent" was exactly what Jamie says, i.e. "if you have this thick slab of something, this is the color you get". Which Jamies math does, and mine doesn't :P 
In my defense, one of my motto's is "I'm not a mathematician - I'm a magician" :)

Reply:
@portsmouth Is this new parameterization for subsurface_bssrdf matching what you had in mind?